source/plugin/data/customdata.rst

```diff
@@ -0,0 +1,297 @@
+After that you will want to create a class that can build the data from a ``DataContainer`` this is known as 
+the :javadoc:`DataBuilder` which can be implemented as follows:
+
+.. code-block:: java
+
+    import org.spongepowered.api.data.persistence.InvalidDataException;
+
+    public class LastAttackerDataBuilder implements DataBuilder<LastAttackerDataSerilizable> {
+    
+        @Override
+        public Optional<LastAttackerDataSerilizable> build(DataView container) throws InvalidDataException {
+            Optional<String> lastAttackerAsStringId container.getString(LastAttackerDataSerilizable.UUID_PATH);
+            if(lastAttackerAsStringId.isPresent()){
+                UUID lastAttacker = UUID.fromString(lastAttackerAsStringId.get());
+                return Optional.of(new LastAttackerDataSerilizable(lastAttacker));
+            }
+            return Optional.empty();
+        }
+    
+    }
+
+
+Registration
+============
+
+Registering your ``DataSerializable`` allows it to be accessible by Sponge and by other plugins in a generic way. The
+game/plugin can create copies of your data and serialize/deserialize your data without referencing any of your classes
+directly.
+
+To register a ``DataSerializable`` Sponge has the :javadoc:`RegisterDataEvent` event. This will allow you to register
+your data with the appropriate ``DataHolder``
+
+Simple Custom Data
+==================
+
+All of above is a lot of work if you just wanting to register a java primitive or ``String`` to
+a ``DataHolder``. Thankfully there is a much shorter way to do all of that. 
+
+.. code-block:: java
+
+    Key<? extends Value<String>> key = Key.from(pluginContainer, "my_simple_data", String.class);
+    DataRegistration myData = DataRegistration.of(key, ServerPlayer.class);
+    event.register(myData);
+
+Registration Key
+================
+
+When it comes to registering your data, you are required to register it with a :javadoc:`Key` which will allow you and
+other developers access to your data manipulator.
+
+
+.. code-block:: java
+
+    import org.spongepowered.api.ResourceKey;
+    import org.spongepowered.api.data.Key;
+    import org.spongepowered.api.data.value.Value;
+
+    ResourceKey resourceKey = ResourceKey(pluginContainer, "last_attacker_manipulator");
+    Key<? extends Value<LastAttackerDataSerilizable>> key = Key
+        .builder()
+        .key(resourceKey)
+        .elementType(LastAttackerDataSerilizable.class)
+        .build();
+
+.. warning::
+
+    Be sure to store your ``Key`` somewhere global so you can access it later.
+
+.. tip::
+
+    You can register a key for a specific element within a DataSerializable
+
+Data Store
+==========
+
+The :javadoc:`DataStore` is used to register your ``Key`` with the appropriate ``DataHolder`` and also register
+any other keys you may have accessing your ``DataSerializable``. In the example below, it creates a ``DataStore``
+and makes it appliciable to only the :javadoc:`Entity` ``DataHolder``.
```

Comment:
Mention how this is to enable the `DataHolder`'s mode of persistence to know where/how to store the data backed by the `Key` in that automatic serialization and deserialization (based on whether it's simple or a DataSerializable kind).